### ISSUE TITLE
Bug in Location Gathering from StackOverflow Service

### ISSUE THREAD

**Issue by rrgayhart** [LookingForMe/lookingfor#37]
If you run `Job.pluck(:location).uniq` you'll find some odd locations. We should investigate how these locations are being created by looking at the job titles and beefing up testing.

Acceptance Criteria:
- Run through bizarre locations - add testing for edge cases
- Beef up regex to handle as many edge cases as possible OR switch to NokoGiri (see Erinna's notes)
- Write one off rake task to correct existing records that are borked
  - www.regex101.com is super helpful

**Comment by PenneyGadget** [comment]
I picked up this issue at the end of my time with this project, so I did not have the time to complete it. I did, however, figure out what the problem is and have some ideas for perhaps how to fix it.

The current regex being used in the `pull_location` method of the `stack_overflow.rb` service is just grabbing the first value found inside parenthesis. If there are 2 sets of parenthesis (i.e. "Developer job (backend) at CoolCompany (Denver, CO)") it will pull "backend" as the location. Just grabbing the _last_ parenthesized value won't work either, as many job listings end with "(allows remote)".

My first thought for a fix was to grab the parenthesized value that also contains a `,` as it seems that all the locations are comma separated (maybe - this is from me looking at a couple hundred records, but we're pulling in thousands). I do think this would fix 90% of the issue, save for listings like "Developer job (Javascript, Ruby, Python) at CoolCompany (Denver, CO)". A thought on that is to create a method that plucks all the language names (which we have in the technology model) and builds a regex that _excludes_ them before we grab the location name. @rrgayhart is a great person to talk to about this, she seems to be a bit of a regex ninja! 

Good luck! :smile_cat:
